### PR TITLE
Illiad and Aeon fields mapping

### PR DESCRIPTION
**Pull request by banukutlu**
Issue #162 and #166

### CODE REVIEW

**Comment by banukutlu** [comment]
> Did you give this a try? Did it not work for some reason? It looks similar to what you have done here.
> 
> https://github.com/traject/traject/blob/a80c58bba8ce9943daf53d0b8691f2d583546cb6/lib/traject/macros/marc21_semantics.rb#L16-L48

tbh I haven't seen this but also the prefixes are a bit different. we are not checking `on` but checking for `OCLC`.

**Comment by cdmo** [comment]
I ran it and it only passed 2/6:

```
Run options: include {:locations=>{"./spec/lib/traject/macros/custom_spec.rb"=>[114]}}
..FFFF

Failures:

  1) Macros spec: extract_oclc_number when 035 field includes ocn maps the oclc number
     Failure/Error: expect(result['oclc_number_display_ssm']).to include('239422053')
       expected ["154806744"] to include "239422053"
     # ./spec/lib/traject/macros/custom_spec.rb:139:in `block in <main>'

  2) Macros spec: extract_oclc_number when 035 field includes ocm maps the oclc number
     Failure/Error: expect(result['oclc_number_display_ssm']).to include('40777018')
       expected ["154806744"] to include "40777018"
     # ./spec/lib/traject/macros/custom_spec.rb:145:in `block in <main>'

  3) Macros spec: extract_oclc_number when 035 field includes OCLC maps the oclc number
     Failure/Error: expect(result['oclc_number_display_ssm']).to include('70197573')
       expected ["154806744"] to include "70197573"
     # ./spec/lib/traject/macros/custom_spec.rb:151:in `block in <main>'

  4) Macros spec: extract_oclc_number when 035 field does not include any of the OCLC prefixes ignores the 035 value
     Failure/Error: expect(result['oclc_number_display_ssm']).to eq %w[154806744 239422053 40777018 70197573]

       expected: ["154806744", "239422053", "40777018", "70197573"]
            got: ["154806744"]

       (compared using ==)
     # ./spec/lib/traject/macros/custom_spec.rb:157:in `block in <main>'

Finished in 0.60376 seconds (files took 3.64 seconds to load)
6 examples, 4 failures

Failed examples:

rspec ./spec/lib/traject/macros/custom_spec.rb:138 # Macros spec: extract_oclc_number when 035 field includes ocn maps the oclc number
rspec ./spec/lib/traject/macros/custom_spec.rb:144 # Macros spec: extract_oclc_number when 035 field includes ocm maps the oclc number
rspec ./spec/lib/traject/macros/custom_spec.rb:150 # Macros spec: extract_oclc_number when 035 field includes OCLC maps the oclc number
rspec ./spec/lib/traject/macros/custom_spec.rb:156 # Macros spec: extract_oclc_number when 035 field does not include any of the OCLC prefixes ignores the 035 value
```

File: spec/lib/traject/macros/custom_spec.rb [hunk 1]
```diff
@@ -110,4 +110,52 @@
...
+    context 'when there is no 035' do
+      it 'does not map record' do
+        @empty_record = MARC::Record.new
+        @empty_record.append(MARC::ControlField.new('001', '000000000'))
+        result = @indexer.map_record(@empty_record)
+        expect(result['oclc_number_display_ssm']).to be_nil
+      end
+    end
+
+    context 'when 035 field includes OCoLC' do
```
Comment (cdmo) [inline spec/lib/traject/macros/custom_spec.rb]:
```suggestion
    context 'when 035 field includes \"OCoLC\"' do
```

File: spec/lib/traject/macros/custom_spec.rb [hunk 2]
```diff
@@ -110,4 +110,52 @@
...
+      it 'maps the oclc number' do
+        expect(result['oclc_number_display_ssm']).to include('154806744')
+      end
+    end
+
+    context 'when 035 field includes ocn' do
```
Comment (cdmo) [inline spec/lib/traject/macros/custom_spec.rb]:
```suggestion
    context 'when 035 field includes \"ocn\"' do
```

File: spec/lib/traject/macros/custom_spec.rb [hunk 3]
```diff
@@ -110,4 +110,52 @@
...
+      it 'maps the oclc number' do
+        expect(result['oclc_number_display_ssm']).to include('239422053')
+      end
+    end
+
+    context 'when 035 field includes ocm' do
```
Comment (cdmo) [inline spec/lib/traject/macros/custom_spec.rb]:
```suggestion
    context 'when 035 field includes \"ocm\"' do
```

File: spec/lib/traject/macros/custom_spec.rb [hunk 4]
```diff
@@ -110,4 +110,52 @@
...
+      it 'maps the oclc number' do
+        expect(result['oclc_number_display_ssm']).to include('40777018')
+      end
+    end
+
+    context 'when 035 field includes OCLC' do
```
Comment (cdmo) [inline spec/lib/traject/macros/custom_spec.rb]:
```suggestion
    context 'when 035 field includes \"OCLC\"' do
```

**Pull request merged**